sled-agent/src/rack_setup/dns_interface.rs

```diff
@@ -0,0 +1,87 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Stubs for the DNS server interface.
```

Comment:
I assume this will wind up changing as a result of #2570.

You might consider just spinning up a real DNS server.  I think that would be pretty straightforward.

Reply:
Does https://github.com/oxidecomputer/omicron/blob/b6d9317bc914f0cd4ea7c33e94bc0d737d492537/sled-agent/src/sim/server.rs#L35 fit here?